src/js/extensions/auto-linker.js

```diff
@@ -0,0 +1,418 @@
+/*global Extension, Util */
+
+var AutoLinker,
+    AutoLinkerStatics = {},
+    KEY_CODES = {
+        SPACE: 32,
+        ENTER: 13
+    },
+    LINK_REGEXP_TEXT;
+
+LINK_REGEXP_TEXT =
+    '(' +
+    // Version of Gruber URL Regexp optimized for JS: http://stackoverflow.com/a/17733640
+    '((?:[a-z][\\w-]+:(?:\\\/{1,3}|[a-z0-9%])|www\\d{0,3}[.]|[a-z0-9.\\-]+[.][a-z]{2,4}\\\/)\\S+(?:[^\\s`!\\[\\]{};:\'\".,?\u00AB\u00BB\u201C\u201D\u2018\u2019]))' +
+    // Addition to above Regexp to support bare domains with common non-i18n TLDs and without www prefix:
+    ')|([a-z0-9\\-]+\\.(com|net|org|edu|gov|mil|aero|asia|biz|cat|coop|info|int|jobs|mobi|museum|name|post|pro|tel|travel|xxx|ac|ad|ae|af|ag|ai|al|am|an|ao|aq|ar|as|at|au|aw|ax|az|ba|bb|bd|be|bf|bg|bh|bi|bj|bm|bn|bo|br|bs|bt|bv|bw|by|bz|ca|cc|cd|cf|cg|ch|ci|ck|cl|cm|cn|co|cr|cs|cu|cv|cx|cy|cz|dd|de|dj|dk|dm|do|dz|ec|ee|eg|eh|er|es|et|eu|fi|fj|fk|fm|fo|fr|ga|gb|gd|ge|gf|gg|gh|gi|gl|gm|gn|gp|gq|gr|gs|gt|gu|gw|gy|hk|hm|hn|hr|ht|hu|id|ie|il|im|in|io|iq|ir|is|it|je|jm|jo|jp|ke|kg|kh|ki|km|kn|kp|kr|kw|ky|kz|la|lb|lc|li|lk|lr|ls|lt|lu|lv|ly|ma|mc|md|me|mg|mh|mk|ml|mm|mn|mo|mp|mq|mr|ms|mt|mu|mv|mw|mx|my|mz|na|nc|ne|nf|ng|ni|nl|no|np|nr|nu|nz|om|pa|pe|pf|pg|ph|pk|pl|pm|pn|pr|ps|pt|pw|py|qa|re|ro|rs|ru|rw|sa|sb|sc|sd|se|sg|sh|si|sj| Ja|sk|sl|sm|sn|so|sr|ss|st|su|sv|sx|sy|sz|tc|td|tf|tg|th|tj|tk|tl|tm|tn|to|tp|tr|tt|tv|tw|tz|ua|ug|uk|us|uy|uz|va|vc|ve|vg|vi|vn|vu|wf|ws|ye|yt|yu|za|zm|zw))';
+
+(function () {
+    'use strict';
+
+    var nodeIteratorDetectsNewNodes = true; // this is the behavior in Chrome, FF, Safari...
+
+    // feature test for the behavior of NodeIterator. In IE11 this didn't behave like the other browsers...
+    (function () {
+        var div = document.createElement('div'),
+            node,
+            it;
+        div.appendChild(document.createTextNode('ab'));
+        div.appendChild(document.createTextNode('c'));
+        it = document.createNodeIterator(div, NodeFilter.SHOW_TEXT, null, false);
+        node = it.nextNode();
+        node.splitText(1);
+        node = it.nextNode();
+        if (node.nodeValue === 'c') {
+            nodeIteratorDetectsNewNodes = false;
+        } else if (node.nodeValue !== 'b') {
+            throw new Error('Unexpected behavior of NodeIterator. Code will not work properly');
```

Comment:
I set up a catch statement in onKeypress now that will disable the event handling if any error/exception is thrown from internal code.